nixos/modules/services/web-apps/glance.nix

```diff
@@ -100,14 +100,64 @@ in
         for more.
 
         Settings containing secret data should be set to an
-        attribute set containing the attribute
-        <literal>_secret</literal> - a string pointing to a file
-        containing the value the option should be set to. See the
-        example in `services.glance.settings.pages` at the weather widget
+        attribute set with this format: `{ _secret = "/path/to/secret"; }`.
+        See the example in `services.glance.settings.pages` at the weather widget
         with a location secret to get a better picture of this.
+
+        Alternatively, you can use a single file with environment variables,
+        see `services.glance.environmentFile`.
       '';
     };
 
+    environmentFile = mkOption {
+      type = types.nullOr types.path;
```

Comment:
Question : is the content of the file going to end up in the nix store when using types.path ?

Reply:
By this module, no: `environmentFile` is only referenced in the systemd service's `EnvironmentFile`, which passes the path of the file and the systemd service will read it at runtime

However the user can still do stupid things like `environmentFile = pkgs.writeText "glance-environment" "XYZ_API_KEY=xxxx"` which will end up in nix store

I don't think the type has any influence over this, and pretty much all other modules set the same type